src/routeGeneration/routeGenerator.ts

```diff
@@ -52,10 +52,9 @@ export class RouteGenerator {
   }
 
   private buildContent(middlewareTemplate: string, pathTransformer: (path: string) => string) {
-    handlebars.registerHelper('json', (context: any) => {
-      return JSON.stringify(context);
-    });
-    const additionalPropsHelper = (additionalProperties: TsoaRoute.ModelSchema['additionalProperties']) => {
+    handlebars.registerHelper('json', (context: any) => JSON.stringify(context));
```

Comment:
Please remove these style changes. There’s nothing wrong with this code functionally. Also please consider that for the entire PR, since I won’t approve a PR that changes code that didn’t need to be changed. And to clarify, code only needs to change if it either fixes a known issue (that has an issue number in tsoa’s github) or if it enhances tsoa (and therefore has a tsoa issue number with the discussed enhancement).

Reply:
(Whoops I see I'm responding to your comments in reverse order, you probably want the read my comment below (the long one) before this.)

I think I already touched upon most of what you're saying here in my long comment below, but just for clarification's sake:

> There’s nothing wrong with this code functionally

No there's nothing wrong *functionally*, but there's a lot of room for easy improvement in readability, accessibility and maintainability.

> Also please consider that for the entire PR, 

As I said in my second comment (from below, I really should've answered these in the correct order...) I'm trying to keep (habitual) *stylistic* changes to a minimum, I'll remove any that snuck in after I'm done.

> since I won’t approve a PR that changes code that didn’t need to be changed.

I'd argue some of these changes are definitely needed (or at least welcome improvements). I don't see how rejecting a change just because it "changes what's already there" helps with improving software. Endless cycles of refactoring are of course undesirable, but a rigorous review & cleanup can be very healthy for a project (at least in my experience).

> And to clarify, code only needs to change if it either fixes a known issue (that has an issue number in tsoa’s github) or if it enhances tsoa (and therefore has a tsoa issue number with the discussed enhancement).

I already touched upon this in my next comment (sorry again).

Reply:
Please understand that I personally wrote the code that you changed. I wrote it that way because it made it shorter horizontally. That’s because I subjectively felt that that they way I wrote the code was better. You have now changed it to what you think is subjectively better. If I was getting paid as part of my day job then I would happily debate with you the merits of breaking code into discrete parts.

Every change that is submitted requires my review. So if a change is submitted for subjective/stylistic reasons, that takes up valuable time of mine that I could be spending reviewing or improving one of the many open issues that tsoa has.

I’m summary, do not change code that does not need to be changed. This was recently reviewed by Endor and WoH and they didn’t have problems with it. We will only accept changes that improve the code objectively. And the only objective improvements that can be made are bug fixes or feature enhancements.

Reply:
> I wrote it that way because it made it shorter horizontally. That’s because I subjectively felt that that they way I wrote the code was better. You have now changed it to what you think is subjectively better.

This is of course true, but it's not like I'm changing it purely because I think it "looks better", none of these changes are. In this specific case I changed it because it avoids an additional (completely useless) variable. This results in more concise code that's easier and faster to understand, because you skip the "where does this variable come from" and "what is this variable used for" steps. It's also more consistent with the rest of the codebase, as well as most other code I've encountered. Honestly, I prefer that to saving a couple characters horizontally.

But there's really no use to arguing about this now, like luke said we need a style guide for things like this so we can make "informed" or "standardized" (not sure what the appropriate word is here) but most importantly consistent choices.

> Every change that is submitted requires my review. So if a change is submitted for subjective/stylistic reasons, that takes up valuable time of mine that I could be spending reviewing or improving one of the many open issues that tsoa has.

While true, I don't think this is a valid reason for rejecting an honest try at improving the codebase. Having a style guide will of course speed up the reviewing process since you can make an objective decision based on this guide.

Also there's another benefit to making the code more contributor friendly: a bigger pool of people that can/want to review PR's. (I'm not saying this PR will immediately solve this issue, I just hope that it opens up the possibility of long term improvements.)

> I’m summary, do not change code that does not need to be changed. [...] We will only accept changes that improve the code objectively. And the only objective improvements that can be made are bug fixes or feature enhancements.

Again, most of my changes are in fact objective improvements (typesafety). Changes like the one you selected here might not be entirely objective but there's definitely an (in my opinion) valid reasoning behind it. As luke said, we need a style guide.
I also want to add a short note that I think that maintainability, contributor-friendliness and readability are in fact objective improvements, but the line you selected here isn't really a good example of that.

> This was recently reviewed by Endor and WoH and they didn’t have problems with it. 

I don't see how "it was approved once" clears it from any kind of improvement. Not all code that's "approved" is perfect (for example due to sloppy reviewing or just not having an optimal solution), I'll take my own shameful #448 as example.
Don't get me wrong on this, I'm not saying that this code is bad or that Endor/WoH haven't reviewed your code properly, I'm just trying to say that I think we should keep an "open mind" about incoming improvements.

Reply:
> In this specific case I changed it because it avoids an additional (completely useless) variable.

For yo that variable is useless. For me it added readability.

There are some people that would prefer to solve an algebra problem as follows:
```
x = 2 * 3 + 4 - 6 * 7
x = -33
```

But I always prefer the following approach. And my algebra teacher recommended this approach as well always saying “show your work”:

```
x = 2 * 3 + 4 - 6 * 7
x = (2 * 3) + 4 - (6 * 7)
x = 5 + 4 - 42
x = -33
```

So again, this is a subjective point. Not one that should cause a change without a bug or an enhancement to give a reason to change it.

Contributions to software should be like surgery. If I have to do surgery, I will fix things related to the problem and that the patient has pre-approved. I don’t just fix the lungs while I’m doing open heart surgery. In the same manner, software contributions should be small, atomic, and should be specific to the issue at hand. So since this line has nothing to do with improving the type safety of the type resolver, then it should not have been changed.

If however there is an open issue related to the route file generation, then I would be more than happy to entertain refactoring on this file. I will indeed have an open mind when surgery on the routeGenerator is necessary.

Reply:
> For yo that variable is useless. For me it added readability.

I may have worded my point of view a little too strong, there's definitely room for debate here. As for the math analogy: I agree with you, but more steps doesn't equal more readability. There's a point where code becomes too convoluted for it's own good. Where exactly that point is should be added in the style guide.
In my opinion this particular piece of code crossed that line, for reasons I'm not going to repeat again. Increased readability/maintainability/accessibility is an enhancement in my opinion, which is why I chose to change it at that particular point in time (which for the record was months ago when I first saw the codebase).

> Contributions to software should be like surgery.

I strongly disagree with this analogy. I think it's easiest to explain my viewpoint with another analogy:
I imagine software more like a computer. If it's not broken you don't want to be constantly changing it, that's just unnecessary hassle. Switching out the parts, verifying everything is still working, installing new drivers, the whole bunch.
However, once in a while you might want to do maintenance to improve your experience. Maybe add a little extra storage, a faster CPU or more RAM. While you're doing these upgrades you might find the cable management to be a little "*lacking*" so to speak, so you decide to clean that up as well.

Even though redoing the cable management might not be much of a tangible enhancement or a fix for a concrete problem you're having right now, it will still definitely benefit you in the long run. Because the next time you want to add some more RAM or switch out a broken hard drive you want to be able to focus on that, instead of having to navigate through or possibly even redo the messy cabling.

> In the same manner, software contributions should be small, atomic, and should be specific to the issue at hand. So since this line has nothing to do with improving the type safety of the type resolver, then it should not have been changed.

I agree with this bit in the sense that I should've split this out more from the start. I messed up in that area, and I'm sorry about that. However, that doesn't mean the changes I'm proposing should be outright rejected.